cirq-google/cirq_google/devices/serializable_device.py

```diff
@@ -108,6 +108,25 @@ def __init__(
         """
         self.qubits = qubits
         self.gate_definitions = gate_definitions
+        self._metadata = cirq.GridDeviceMetadata(
+            [
+                (pair[0], pair[1])
+                for gate_defs in gate_definitions.values()
+                for gate_def in gate_defs
+                if gate_def.number_of_qubits == 2
+                for pair in gate_def.target_set
+                if len(pair) == 2 and pair[0] < pair[1]
+            ],
+            cirq.Gateset(
+                *[g for g in gate_definitions.keys() if isinstance(g, (cirq.Gate, type(cirq.Gate)))]
+            ),
+            None,
```

Comment:
I think the gate durations exist currently, so I think we could populate this if we wanted to.

Reply:
We could, but I think it's best left for when we can turn 100% of our attention to cleaning up things for `SerializableDevice` this PR is a means to an end for getting cleaner deprecations happening in `cirq-core` for `qid_pairs`.